octoprint_discordremote/command.py

```diff
@@ -45,6 +53,90 @@ def parse_command(self, string):
         else:
             return command['cmd']()
 
+    def get_timelapse(self, params):
+        if len(params) > 2:
+            return None, error_embed(title='Too many parameters',
+                                     description='Should be: /gettimelapse [filename]')
+        if shared_vars.base_url is None or shared_vars.base_url == "":
+            return None, error_embed(title="Base URL Setting",
+                                     description="Check the Base URL setting in the settings dialog. It may be incorrectly set."
+                                                 "\nbase_url: " + str(shared_vars.base_url))
+
+        api_key = self.plugin.get_settings().global_get(["api", "key"])
+        port = self.plugin.get_settings().global_get(["server", "port"])
+        header = {'X-Api-Key': api_key}
+
+        response = requests.get("http://127.0.0.1:%s/api/timelapse" % port, headers=header)
+        data = response.json()
+
+        for x in data['files']:
+            if x['name'] == params[1]:
+                return None, info_embed(title=params[1], description=("http://" + str(shared_vars.base_url) + x['url']))
+        return None, error_embed(title="File Not Found", description=params[1])
+
+    def get_file(self, params):
+        if len(params) > 3:
+            return None, error_embed(title='Too many parameters',
+                                     description='Should be: /getfile [location] [filename]. Location is either local or sdcard.')
+        if shared_vars.base_url is None or shared_vars.base_url == "":
+            return None, error_embed(title="Base URL Setting",
+                                     description="Check the Base URL setting in the settings dialog. It may be incorrectly set.")
+
+        api_key = self.plugin.get_settings().global_get(["api", "key"])
+        port = self.plugin.get_settings().global_get(["server", "port"])
+        header = {'X-Api-Key': api_key}
+        url = "http://" + str(shared_vars.base_url) + "/api/files/" + params[1] + "/" + params[2]
+        try:
+            response = requests.get(
+                url,
+                headers=header)
+            if not response:
+                return None, error_embed(title="ConnectionError", description="URL: " + str(url))
+        except ConnectionError:
+            return None, error_embed(title="ConnectionError", description="URL: " + str(url))
+        data = response.json()
+        if response.status_code == 200:
+            return None, info_embed(title=params[2], description=(data['refs'])['download'])
+        return None, error_embed(title="File Not Found", description=params[2])
+
+    def timelapse(self):
```

Comment:
@OctopusProteins Seems that the /gettimelapse and /timelapse functionality cross over a bit, i suggest merging them together. Just get rid of /gettimelapse, and add the URL into the /timelapse function?